staking/programs/staking/src/lib.rs

```diff
@@ -706,4 +706,28 @@ pub mod staking {
             Some(ctx.accounts.config.agreement_hash);
         Ok(())
     }
+
+    /** Recovers a user's `stake account` ownership by transferring ownership
+     * from a token account to the `owner` of that token account.
+     *
+     * This functionality addresses the scenario where a user mistakenly
+     * created a stake account using their token account address as the owner.
+     */
+    pub fn recover_account(ctx: Context<RecoverAccount>) -> Result<()> {
+        // Check that there aren't any positions (i.e., staked tokens) in the account.
+        // Transferring accounts with staked tokens might lead to double voting
+        require!(
+            ctx.accounts.stake_account_metadata.next_index == 0,
+            ErrorCode::SplitWithStake
```

Comment:
It's weird you created a new error `RecoverWithStake` but you aren't using it here.

Reply:
I fixed this i https://github.com/pyth-network/governance/pull/426